flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/metadata/MetadataV6Serializer.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.checkpoint.metadata;
+
+import org.apache.flink.annotation.Internal;
+import org.apache.flink.annotation.VisibleForTesting;
+import org.apache.flink.runtime.checkpoint.StateObjectCollection;
+import org.apache.flink.runtime.state.InputStateHandle;
+import org.apache.flink.runtime.state.OutputStateHandle;
+
+import javax.annotation.Nullable;
+
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+
+/** V6 serializer that use ChannelStateHandleSerializerV2. */
+@Internal
+public class MetadataV6Serializer extends MetadataV5Serializer {
+
+    public static final MetadataSerializer INSTANCE = new MetadataV6Serializer();
+
+    public static final int VERSION = 6;
+
+    private final ChannelStateHandleSerializer channelStateHandleSerializer =
+            new ChannelStateHandleSerializerV2();
```

Comment:
How about adding a protect method `get ChannelStateHandleSerializer()` and derived in `MetadataV6Serializer`? IIUC the logic of these methods for channelState are identical, only difference in the serializer?

Reply:
Thank you for this good suggestion @Zakelly  ! I have update the PR according to this suggestion, and the CI has passed. Please take a look.